gen/gen_comps.py

```diff
@@ -0,0 +1,63 @@
+import glob
+import os
+import sys
+
+from mako.template import Template
+
+
+def find_matching_stub(body_field, stubs):
+    for x in stubs:
+        if x['name'] == body_field['name']:
+            return x
+    return None
+
+
+def main():
+    files = [f for f in glob.glob('gen/comp/*')]
+    comp_names = sorted([os.path.basename(f) for f in files])
+
+    component_managers_header_template = Template(filename='gen/templates/component_managers.h.gen')
+    component_header_template = Template(filename='gen/templates/comp_component.h.gen')
+    component_impl_template = Template(filename='gen/templates/comp_component.cc.gen')
+
+    print("Genning header: src/component/component_managers.h")
+    with open("src/component/component_managers.h", "w") as g:
+        g.write(component_managers_header_template.render(comp_names=comp_names))
```

Comment:
I'd go one step further and build the whole model first -- you're about to want another output like `component_managers.h`, but which needs to know internals of each component.